lib/wallet/wallet.js

```diff
@@ -1606,15 +1607,15 @@ class Wallet extends EventEmitter {
    * @param {String} name
    * @param {Number} value
    * @param {Number} lockup
-   * @param {Number} acct
+   * @param {Number|String} acct
    * @returns {MTX}
    */
 
   async makeBid(name, value, lockup, acct) {
     assert(typeof name === 'string');
     assert(Number.isSafeInteger(value) && value >= 0);
     assert(Number.isSafeInteger(lockup) && lockup >= 0);
-    assert((acct >>> 0) === acct);
+    assert((acct >>> 0) === acct || typeof acct === 'string');
```

Comment:
The alternative to making `acct` polymorphic is doing a database lookup earlier to convert the human readable account name to the account index integer that this function is expecting. The line

```javascript
const addr = await this.receiveAddress(acct);
```

works on either an integer or a string, where the integer is expected to be the account index or the string is expected to be the human readable account name.

Reply:
I think this current approach works fine. The issue we are solving is that validation of the RPC and HTTP endpoints, only accept strings:
- https://github.com/handshake-org/hsd/pull/163/files/96945ed56e8580ca28346850f887261a972a8102#diff-e71441ed6076bfec17d8b3e3e6a9cd8dR1601
- https://github.com/handshake-org/hsd/pull/145/files#diff-fd026b8e2fb746e2c72b16a555745ca6R1990

When the account option is not passed through the API, the `acct` parameter defaults to 0.
- https://github.com/tynes/hsd/blob/wallet-http-create/lib/wallet/wallet.js#L1550

Currently, I believe this is the only way this function would ever receive an integer as its `acct` parameter.